### PR TITLE
Fix for network test failing randomly

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: core/network/src/protocol/sync.rs [hunk 1]
```diff
@@ -198,7 +198,9 @@ pub struct Status<B: BlockT> {
 	/// Target sync block number.
 	pub best_seen_block: Option<NumberFor<B>>,
 	/// Number of peers participating in syncing.
-	pub num_peers: u32
+	pub num_peers: u32,
```
Comment (bkchr) [inline core/network/src/protocol/sync.rs]: Why are these values `u32` and not `usize`?

File: core/network/src/test/mod.rs [hunk 2]
```diff
@@ -591,6 +591,9 @@ pub trait TestNetFactory: Sized {
 		// Return `NotReady` if there's a mismatch in the highest block number.
 		let mut highest = None;
 		for peer in self.peers().iter() {
+			if peer.is_major_syncing() || peer.network.num_queued_blocks() != 0 {
+				return Async::NotReady
```
Comment (bkchr) [inline core/network/src/test/mod.rs]: Is that really okay? Are we sure that something wakes up the task to continue?

Comment (arkpar) [reply core/network/src/test/mod.rs]: Queue events wake up the task I believe.

File: core/network/src/protocol/sync.rs (hunk 1 above)
Comment (arkpar) [reply core/network/src/protocol/sync.rs]: Because they don't depend on machine pointer size. `u32` is more than enough.

**Pull request merged**